### PR TITLE
Fix for failing export-openapi.sh on K8s 1.19

### LINKED ISSUE #2159: export-openapi.sh doesn't work with Kubernetes 1.19
**What happened**: Running `make gen-embedded-openapi` exits with the error `jq: error (at <stdin>:0): Cannot index array with string "spec"`

**What you expected to happen**: The script to complete successfully.

**How to reproduce it (as minimally and precisely as possible)**: 

I also tried `make shell` and then ran the first two lines of the script:

```
kubectl proxy &
curl http://127.0.0.1:8001/openapi/v2 | jq 'del(.. | .["x-kubernetes-patch-strategy"]?, .["x-kubernetes-patch-merge-key"]?, .["x-kubernetes-list-type"]?)' | jq 'del(.. |  .["x-kubernetes-group-version-kind"]?, .["x-kubernetes-list-map-keys"]? )'
```

and it gave the same error.

**Anything else we need to know?**:

**Environment**:
- Agones version: HEAD
- Kubernetes version (use `kubectl version`): 1.19
- Cloud provider or hardware configuration: GKE
- Install method (yaml/helm): `make gcloud-test-cluster`
- Troubleshooting guide log(s):
- Others:

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking

/kind bug

> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:

Some new documentation in 1.19 meant that there was some extra work to be done to escape certain strings properly.

This does that, and also provides the updated output for `PodTemplateSpec` and `ObjectMeta`.

**Which issue(s) this PR fixes**:

Work on #2129
Fixes #2159

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by markmandel** [comment]
Oops forgot to regen install.yaml

**Comment by markmandel** [comment]
Oh yeah, helm fails to upgrade on crd change. wheeee.

File: build/README.md [hunk 1]
```diff
@@ -574,8 +574,10 @@ Compile the ping binary and then build the docker image
 #### `make gen-install`
 Generate the `/install/yaml/install.yaml` from the Helm template
 
-#### `gen-embedded-openapi`
-Generate the embedded OpenAPI specs for existing Kubernetes Objects, such as `PodTemplateSpec` and `ObjectMeta`
+#### `make gen-embedded-openapi`
+Generate the embedded OpenAPI specs for existing Kubernetes Objects, such as `PodTemplateSpec` and `ObjectMeta`.
+
+This should be run against a clean or brand new cluster, as external CRD's or schemas could cause errors to occur.
```
Comment (roberthbailey) [inline build/README.md:580]: Is there a build target to do this? I found that when I created clusters manually that the kubectl proxy command in the script couldn't find them but when I created a cluster with `make gcloud-test-cluster ` it was able to connect.

**Review by roberthbailey — APPROVED** [review]

File: build/README.md (hunk 1 above)
Comment (markmandel) [reply build/README.md:580]: `make gcloud-test-cluster` does work - although admittedly that will install Agones (which I found out the hard way, I did a `helm uninstall` on it) - so there isn't a make target that makes an empty cluster.

Comment (roberthbailey) [reply build/README.md:580]: Thanks. I've updated the upgrade checklist to include the helm uninstall command.

**Comment by roberthbailey** [comment]
Build error:

```
# if IMAGE_PULL_SECRET_FILE is specified, create the agones-system namespace and install the secret
\
	helm upgrade --install --atomic --namespace=agones-system \
	--create-namespace \
	--set agones.image.tag=1.16.0-3d33d94,agones.image.registry=gcr.io/agones-images \
	--set agones.image.controller.pullPolicy="Always",agones.image.sdk.alwaysPull=true \
	--set agones.image.controller.pullSecret= \
	--set agones.ping.http.serviceType="LoadBalancer",agones.ping.udp.serviceType="LoadBalancer" \
	--set agones.allocator.http.serviceType="LoadBalancer" \
	--set agones.controller.logLevel="debug" \
	--set agones.crds.cleanupOnDelete=true \
	--set agones.featureGates="PlayerTracking=true&SDKWatchSendOnExecute=false&RollingUpdateOnReady=false&NodeExternalDNS=true" \
	--set agones.allocator.http.loadBalancerIP=35.227.176.137 \
	 \
	agones /go/src/agones.dev/agones/install/helm/agones/
Error: UPGRADE FAILED: release agones failed, and has been rolled back due to atomic being set: cannot patch "fleets.agones.dev" with kind CustomResourceDefinition:  "" is invalid: patch: Invalid value: ... cannot convert int64 to float64
make: *** [install] Error 1
```

**Comment by roberthbailey** [comment]
It looks like this is the same error as we document on https://agones.dev/site/docs/installation/upgrading/#installation-with-helm which will hopefully be resolved with the next k8s upgrade (to 1.20). 

I've removed the helm installation and restarted the tests.

**Comment by roberthbailey** [comment]
I'm getting all sorts of fun errors tonight. This failure was with gsutil:

```
Already have image (with digest): gcr.io/cloud-builders/gsutil
Copying file://agonessdk-server-1.16.0-3d33d94.zip [Content-Type=application/zip]...
ResumableUploadAbortException: Upload complete with 1536 additional bytes left in stream; this can happen if a file changes size while being uploaded
CommandException: 1 file/object could not be transferred.
/ [0/1 files][    0.0 B/133.4 MiB]   0% Done   
```

**Pull request merged**